truss/remote/baseten/core.py

```diff
@@ -133,15 +133,29 @@ def create_truss_service(
             f"truss=={truss.version()}",
             is_trusted,
         )
-    else:
+
+        return (model_version_json["id"], model_version_json["version_id"])
+
+    if model_id is None:
         model_version_json = api.create_model_from_truss(
-            model_name,
-            s3_key,
-            config,
-            semver_bump,
-            f"truss=={truss.version()}",
-            is_trusted,
-            model_id,
+            model_name=model_name,
+            s3_key=s3_key,
+            config=config,
+            semver_bump=semver_bump,
+            client_version=f"truss=={truss.version()}",
+            is_trusted=is_trusted,
         )
-
-    return (model_version_json["id"], model_version_json["version_id"])
+        return (model_version_json["id"], model_version_json["version_id"])
+
+    # Case where there is a model id already, create another version
+    model_version_json = api.create_model_version_from_truss(
+        model_id=model_id,
+        s3_key=s3_key,
+        config=config,
+        semver_bump=semver_bump,
+        client_version=f"truss=={truss.version()}",
+        is_trusted=is_trusted,
+    )
+    model_version_id = model_version_json["id"]
+    model_version = api.get_model_version_by_id(model_version_id)
```

Comment:
you don't need to do this. You already have model_id on line 152

Reply:
good point